base/memory.hpp

```diff
@@ -0,0 +1,31 @@
+// Copyright 2017 Husky Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <sys/sysinfo.h>
+#include <sys/types.h>
+
+namespace husky {
+namespace base {
+class Memory {
+   public:
+    static int64_t total_phys_mem();
```

Comment:
Reminds of something. It's better to use `size_t` instead of `int64_t`? Can you check?